tests/std/tests/P0645R10_text_formatting_grapheme_clusterization/test.cpp

```diff
@@ -0,0 +1,2053 @@
+    {{U'\x378', U'\x600'}, {0, 1, 2}}, {{U'\x378', U'\x308', U'\x600'}, {0, 2, 3}}, {{U'\x378', U'\x903'}, {0, 2}},
+    {{U'\x378', U'\x308', U'\x903'}, {0, 3}}, {{U'\x378', U'\x1100'}, {0, 1, 2}},
+    {{U'\x378', U'\x308', U'\x1100'}, {0, 2, 3}}, {{U'\x378', U'\x1160'}, {0, 1, 2}},
+    {{U'\x378', U'\x308', U'\x1160'}, {0, 2, 3}}, {{U'\x378', U'\x11a8'}, {0, 1, 2}},
+    {{U'\x378', U'\x308', U'\x11a8'}, {0, 2, 3}}, {{U'\x378', U'\xac00'}, {0, 1, 2}},
+    {{U'\x378', U'\x308', U'\xac00'}, {0, 2, 3}}, {{U'\x378', U'\xac01'}, {0, 1, 2}},
+    {{U'\x378', U'\x308', U'\xac01'}, {0, 2, 3}}, {{U'\x378', U'\x231a'}, {0, 1, 2}},
+    {{U'\x378', U'\x308', U'\x231a'}, {0, 2, 3}}, {{U'\x378', U'\x300'}, {0, 2}},
+    {{U'\x378', U'\x308', U'\x300'}, {0, 3}}, {{U'\x378', U'\x200d'}, {0, 2}},
+    {{U'\x378', U'\x308', U'\x200d'}, {0, 3}}, {{U'\x378', U'\x378'}, {0, 1, 2}},
+    {{U'\x378', U'\x308', U'\x378'}, {0, 2, 3}}, {{U'\xd', U'\xa', U'\x61', U'\xa', U'\x308'}, {0, 2, 3, 4, 5}},
+    {{U'\x61', U'\x308'}, {0, 2}}, {{U'\x20', U'\x200d', U'\x646'}, {0, 2, 3}},
+    {{U'\x646', U'\x200d', U'\x20'}, {0, 2, 3}}, {{U'\x1100', U'\x1100'}, {0, 2}},
+    {{U'\xac00', U'\x11a8', U'\x1100'}, {0, 2, 3}}, {{U'\xac01', U'\x11a8', U'\x1100'}, {0, 2, 3}},
+    {{U'\x1f1e6', U'\x1f1e7', U'\x1f1e8', U'\x62'}, {0, 2, 3, 4}},
+    {{U'\x61', U'\x1f1e6', U'\x1f1e7', U'\x1f1e8', U'\x62'}, {0, 1, 3, 4, 5}},
+    {{U'\x61', U'\x1f1e6', U'\x1f1e7', U'\x200d', U'\x1f1e8', U'\x62'}, {0, 1, 4, 5, 6}},
+    {{U'\x61', U'\x1f1e6', U'\x200d', U'\x1f1e7', U'\x1f1e8', U'\x62'}, {0, 1, 3, 5, 6}},
+    {{U'\x61', U'\x1f1e6', U'\x1f1e7', U'\x1f1e8', U'\x1f1e9', U'\x62'}, {0, 1, 3, 5, 6}},
+    {{U'\x61', U'\x200d'}, {0, 2}}, {{U'\x61', U'\x308', U'\x62'}, {0, 2, 3}},
+    {{U'\x61', U'\x903', U'\x62'}, {0, 2, 3}}, {{U'\x61', U'\x600', U'\x62'}, {0, 1, 3}},
+    {{U'\x1f476', U'\x1f3ff', U'\x1f476'}, {0, 2, 3}}, {{U'\x61', U'\x1f3ff', U'\x1f476'}, {0, 2, 3}},
+    {{U'\x61', U'\x1f3ff', U'\x1f476', U'\x200d', U'\x1f6d1'}, {0, 2, 5}},
+    {{U'\x1f476', U'\x1f3ff', U'\x308', U'\x200d', U'\x1f476', U'\x1f3ff'}, {0, 6}},
+    {{U'\x1f6d1', U'\x200d', U'\x1f6d1'}, {0, 3}}, {{U'\x61', U'\x200d', U'\x1f6d1'}, {0, 2, 3}},
+    {{U'\x2701', U'\x200d', U'\x2701'}, {0, 3}}, {{U'\x61', U'\x200d', U'\x2701'}, {0, 2, 3}}};
+
+template <>
+const test_case_data<char> test_data<char>[602] = {{{static_cast<char>(0x20), static_cast<char>(0x20)}, {0, 1, 2}},
+    {{static_cast<char>(0x20), static_cast<char>(0xcc), static_cast<char>(0x88), static_cast<char>(0x20)}, {0, 2, 3}},
+    {{static_cast<char>(0x20), static_cast<char>(0xd)}, {0, 1, 2}},
+    {{static_cast<char>(0x20), static_cast<char>(0xcc), static_cast<char>(0x88), static_cast<char>(0xd)}, {0, 2, 3}},
+    {{static_cast<char>(0x20), static_cast<char>(0xa)}, {0, 1, 2}},
+    {{static_cast<char>(0x20), static_cast<char>(0xcc), static_cast<char>(0x88), static_cast<char>(0xa)}, {0, 2, 3}},

```

Comment:
why not replace `static_cast<char>(0xcc)` with `'\xcc'` or via user-defined literals to make it less verbose

Reply:
'\xcc' should work fine, to be honest I just picked static_cast somewhat arbitrarily. Using a UDL is overkill considering this is generated code.